core/client/src/main/java/alluxio/client/ClientContext.java

```diff
@@ -42,8 +41,12 @@
    * This method requires that configuration has been initialized.
    */
   public static void init() {
-    String masterHostname =
-        Preconditions.checkNotNull(Configuration.get(PropertyKey.MASTER_HOSTNAME));
+    String masterHostname;
+    if (Configuration.containsKey(PropertyKey.MASTER_HOSTNAME)) {
+      masterHostname = Configuration.get(PropertyKey.MASTER_HOSTNAME);
+    } else {
+      masterHostname = NetworkAddressUtils.getLocalHostName();
```

Comment:
this is needed by tests. I've added a warning message if we hit this branch outside of tests.